### ISSUE TITLE
How can we support continue training?

### ISSUE THREAD

**Issue by ja-thomas** [mlr-org/mlr3learners#7]
This probably rather belongs in mlr3, but there needs to be custom code for learners that support that as well.

For a lot of algorithms (e.g. boosting, gradient descent based algos), training can be continued and models can be updated. 

This was often wanted by people for mlr (especially in combination with early stopping) is there a nice way to support this in a general way for algorithms that support this?

**Comment by mllg** [comment]
My proposal:

You trigger the initial model fit with the `$train()` method of the experiment `e`. This stores the model in `e$learner$model`. You could now update the hyperparameters of the learner (e.g., increase the number of boosting iterations) and write a function `learner$continue()` which checks (a) if it is feasible to do some more iterations, and (b) updates `learner$model`.

**Comment by mllg** [comment]
NB: We could also do this for ranger in a future version (https://github.com/imbs-hl/ranger/issues/356).

**Comment by mllg** [comment]
@berndbischl Your 2 cents?

**Comment by mllg** [comment]
Hm, this does not allow to tune efficiently though. We would need to find a way to to tell the tuner that some hyperparameters are special.

**Comment by berndbischl** [comment]
first of all, supporting this + efficient tuning would be a great feature. it would repair something that mlr2 did until now very suboptimally.

**Comment by ja-thomas** [comment]
With paradox we can add flags to parameters, correct? So we could indicate which parameters are relevant for retraining/continue training. This combined with a learner attribute should make it quite easy right?

**Comment by ja-thomas** [comment]
Also related, but slightly different:

How (if at all) could we support continue training on new/changed dataset?
This will probably be much harder

**Comment by berndbischl** [comment]
i dont think we should discuss this further NOW. we dont have any chance to resolve this until the end of th week. but me me this is a higher prio issue, to strengthen the API

**Comment by XavierPrudent** [comment]
Dear all,
Such a feature would be a major asset for your package. Especially as in the scope of operationability, a continuous training is often required.
Regards,
Xavier

**Comment by mllg** [comment]
See https://github.com/mlr-org/mlr3/issues/272.
